service/http/config.go

```diff
@@ -66,11 +66,8 @@ func (cfg *HTTP2Config) InitDefaults() error {
 
 // SSLConfig defines https server configuration.
 type SSLConfig struct {
-	// Port to listen as HTTPS server, defaults to 443.
-	Port int
-
-	// Redirect when enabled forces all http connections to switch to https.
-	Redirect bool
+	// Socket to handle as https server.
+	Address string
```

Comment:
This will break config for existing users.

Reply:
Yes, I have to do so if there's no better solution, thus, it's a breaking-change.

If the HTTP service is on unix socket, not TCP socket, we could not achieve such 'redirection', because:

 * unix sockets do not have address neither port, we could not simply keep the address and change the port to the one specified in SSL block in config file.
 * it's very likely that unix sockets are used with a reverse proxy (e.g. apache httpd, nginx), the redirection can be done by the reverse proxy. (The reverse proxy may also act as a SSL terminator.)

Reply:
I think it's better to address this PR to the HTTP plugin in the #release_2.0 branch. Or, if you don't mind, I may create a new PR to the #release_2.0 branch and add you as a reviewer.

Reply:
While I understand the logic - this WILL break all existing installations of RR. Also, redirect to the HTTPS version does not require knowledge of the listening socket, it's enough (in theory) to simply change the schema.